src/core/pressure_inline.hpp

```diff
@@ -201,8 +196,10 @@ inline void add_kinetic_virials(Particle const &p1, bool v_comp) {
 
   /* kinetic energy */
   if (v_comp) {
-    obs_scalar_pressure.local.kinetic[0] +=
-        (p1.m.v - (p1.f.f * (0.5 * time_step / p1.p.mass))).norm2() * p1.p.mass;
+    // velocity at half the time step: v(t + dt / 2) =
```

Comment:
Since this is purely additive, and only needed for NpT (IIRC), shouldn't that be added in the NpT calculation? The `time_step` in the observable calculation is really alien to me, and most users of this function would not be concerned with this.

Reply:
Not sure if I understand what you mean. The velocity is subtracted, then squared before being accumulated.

Also, the NpT scalar pressure diverges from the pressure tensor a few line below, since we use uncorrected velocities. This will have to change once we remove the scalar pressure observable in favor of the trace of the pressure tensor.

Reply:
Sorry, I meant that if the logic for the calculation should be reused with multiple velocities it should be a free function that takes the velocity as an argument, which can then be called in the pressure calc with the particle velocity, and in NpT with the corrected velocity. The oddness of the function stems from the fact that it does two things (obtain the velocity, and then do the calculation), which hampers the reusability of the latter. But if this drops out anyway, we can leave it as is.

Reply:
I just tried it in 399ef52805d, but it fails the `PressureTensor` test because velocity compensation is done by default by `PressureTensor`, even when NpT isn't active. We could add a parameter `v_comp` to `PressureTensor`, in which case `Pressure` and `Energy` should also get it, although `Energy` will not use it (it doesn't have the logic to compensate the linear velocity and rotational velocity). I'd suggest to leave things as they are for now; once we've got a plan to create all the observables, we'll calculate the scalar pressure as the trace of the pressure tensor and this `v_comp` variable will be unused. This will break the NpT pressure calculation, but honestly the NpT pressure/energy calculation is already questionable (the NpT PV=NRT sample was working correctly according to Michael 2 years ago, but it was broken when I and Ingo used it 1 year ago).